### PR TITLE
Geometry, Cash Register and Credit Card Validator - Taryn Ewens

### PR DESCRIPTION
**Pull request by t33dev**
Difficulty 9/10 always

Struggled with finding the obtuse triangle - what is math? The education system has failed me!
I know see that I did the cash register in a very long winded way.
Struggled in 2 parts of the credit card validator...
- couldn't figure out how to find if all characters are the same. Tried to make a loop with Joels suggestion, inside the else/if statements and couldn't get it working.
- couldn't figure out how to find the sum of the numbers in the string 

Can loops be made inside if/else statements? 
Explanation of Joels tip: "you could loop through the string and return true/set a `differenceFound` flag to true as soon as you find that `str[i] !== str[i + 1]` (edited)" |

Took 5 and a half hours

### CODE REVIEW

**Comment by johnofsydney** [comment]
Taryn, I have a feeling you will be very successful at this course - you are giving it 100%, and the inescapable result will be that you will know the material very well.

This part is supposed to be long winded, because often that's the easiest way to understand. And there is no substitute for hours at the coal face.

To your comments:
Maths - who cares? not me. The point of this exercise is not the maths. it's the creation of functions and the passing and returning of variables
Cash register - can you come see me and explain: 
"let sum = newArray.reduce( (prev, curr) => prev + curr )"

credit card - the part about all chars the same is prolly the hardest bit. The str[i] !== str[i + 1] only checks for consecutive numbers, better I think to check for str[0] !== str[i]

As for summing the digits - I hope my code along helped, otherwise, come see me and I can talk you through it.

For this Credit Card function - your code is easy to read, your logic is excellent - even though you couldnt get each bit you needed to, you identified what problems you needed to solve, which is more than half of the problem


To summarise - you are doing a great job - keep it up!

With your geometry functions, you need to be passing in  a variable, can you come see me about this - I think I can show you a quick easy example.
